src/frequenz/channels/timer.py

```diff
@@ -497,6 +498,13 @@ def __init__(  # noqa: DOC503 pylint: disable=too-many-arguments
             start_delay: The delay before the timer should start. If `auto_start` is
                 `False`, an exception is raised. This has microseconds resolution,
                 anything smaller than a microsecond means no delay.
+            tick_at_start: When `True`, the timer will trigger immediately after
+                starting, and then wait for the interval before triggering
+                again.  When `False`, the timer will wait the interval before
+                triggering for the first time.  If `auto_start` is `False` and
+                this is `True`, an exception is raised.  If a `start_delay` is
```

Comment:
I don't get why raising if `auto_start` is `False`? What is the issue with making this first tick delayed to when the timer actually starts, like when using `start_delay`?

Reply:
Restricted by how it is implemented.  But if they need it, users can specify `tick_on_start=True` in the call to `reset`, just like with `start_delay`.

Reply:
> What is the issue with making this first tick delayed to when the timer actually starts, like when using `start_delay`?

No, that's not how `start_delay` is implemented.  It also doesn't work with `auto_start=False`.

Reply:
Mmm, OK.